### PR TITLE
Api add

### PR DESCRIPTION
**Pull request by tslekwadi**

### CODE REVIEW

**Comment by wvictor-rr** [comment]
Hey, the project appears to be using .Net Framework, not .Net Core, and is based on an MVC template rather than a WebApi project template.

Is this intentional?

**Review by cilliemalan — DISMISSED** [review]
Wimpy is going to help you start over.

This is the wrong .Net

**Review by wvictor-rr — CHANGES_REQUESTED** [review]
The project was not set up correctly. I'll help you guys redo it.

In the short term this is fine to do your demo with

**Comment by wvictor-rr** [comment]
Some points on the requirements for this project

- The backend must be done using .Net Core 2.0
- The project template to use is **WebAPI** and not **MVC**

File: Bunny101APICore/Bunny101APICore/Controllers/ChatsController.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            db.ChatMessages.Add(message);
+
+            db.SaveChanges();
+
+            return message.Id;
+        }
+
+        [HttpGet("{id}")]
+        [Route("GetConversation")]
+        public List<ChatMessage> Get(string r, string s)
```
Comment (grootendorste) [inline Bunny101APICore/Bunny101APICore/Controllers/ChatsController.cs]: Please comment on what the parameters are or give them better names.

**Review by grootendorste — CHANGES_REQUESTED** [review]
Not bad, just one thing.

File: Bunny101APICore/Bunny101APICore/Controllers/ChatsController.cs (hunk 1 above)
Comment (tslekwadi) [reply Bunny101APICore/Bunny101APICore/Controllers/ChatsController.cs]: ok

File: Bunny101APICore/Bunny101APICore/appsettings.Development.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿{
+  "Logging": {
```
Comment (wvictor-rr) [inline Bunny101APICore/Bunny101APICore/appsettings.Development.json:2]: Preferably put the connection string in the settings. This way you can connect to different databases between environments (When or If you guys have a remote testing server)
eg.
```
"ConnectionStrings": {
    "DefaultConnection": "Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=MessagesDB;Integrated Security=True;Connect Timeout=30;"
  },
```

File: Bunny101APICore/Bunny101APICore/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+            services.AddCors();
+
+            services.AddSingleton(Configuration);
+            services.AddScoped<ChatsController>();
+
+            var connection = @"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=MessagesDB;Integrated Security=True;Connect Timeout=30;";
+            services.AddDbContext<ApplicationDBContext>(options => options.UseSqlServer(connection));
```
Comment (wvictor-rr) [inline Bunny101APICore/Bunny101APICore/Startup.cs]: Once your connection string is in the settings file, use this instead

```
options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection"))
```

**Review by wvictor-rr — CHANGES_REQUESTED** [review]
Good work on changing the project.

I just require some small changes

File: Bunny101APICore/Bunny101APICore/Controllers/ChatsController.cs (hunk 1 above)
Comment (wvictor-rr) [reply Bunny101APICore/Bunny101APICore/Controllers/ChatsController.cs]: 👍

**Review by wvictor-rr — APPROVED** [review]

**Review by retrotshepo — APPROVED** [review]

**Pull request merged**